test/container_manager/container_manager.go

```diff
@@ -0,0 +1,62 @@
+package container_manager
+
+import (
+	"context"
+	"github.com/testcontainers/testcontainers-go"
+	"log"
+	"os"
+	"time"
+)
+
+
+type ContainerManager interface {
+	StartKafkaZookeeperContainers()
+	StopKafkaZookeeperContainers()
+	StartCanary()
+}
+
+
+type controller struct {
+	ctx       context.Context
+	container testcontainers.Container
+	kafkaContainer *testcontainers.LocalDockerCompose
+	isDefaultNetworkRunning bool
+	host string
+}
+
+
+func (c *controller) setContext() {
+	c.ctx = context.Background()
+}
+
+
+func (c *controller) StartKafkaZookeeperContainers() {
+	dir, err := os.Getwd()
+	if err != nil {
+		log.Fatal(err)
+	}
+	path := dir + "/compose-kafka-zookeeper.yaml"
+	c.kafkaContainer = testcontainers.NewLocalDockerCompose([]string{path}, "someName")
+	c.kafkaContainer.WithCommand([]string{"up", "-d"}).Invoke()
+
+	log.Println("docker compose network created")
+}
+
+
+func (c *controller) StopKafkaZookeeperContainers() {
+	c.kafkaContainer.Down()
+}
+
+
+func CreateManager() ContainerManager {
+	var cManipulation ContainerManager = &controller{}
+	return cManipulation
+}
+
+
+func (c *controller) StartCanary() {
+	log.Println("starting canary")
+	go createCanary()
```

Comment:
if we are doing an end-to-end test, as we start ZK and K as containers shouldn't be do kind of same thing with canary? Maybe not container but local application but not re-writing the canary main logic for starting all the services (you already mentioned the DRY principle).

Reply:
Would be possible to do something similar as we have in the bridge tests? Run canary locally or somewhere and work with it?

Reply:
Something like that is my original solution, but methods which are used to start canary are in the main package, so in Go I have no other choice than Copy Paste, so instead of it I start Canary separately via go cmd. line (Exec).